File: ffmpegdemo/ffmpeg2/src/main/java/com/example/ffmpeg2/MainActivity.java
package com.example.ffmpeg2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    static {
        System.loadLibrary("native-lib");
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toast.makeText(this, "" + stringFromJNI(), Toast.LENGTH_SHORT).show();
    }

         //native String stringFromJNI();
    public native String stringFromJNI();
}

File: 源码/gradle/hotfix/app/src/main/java/com/imooc/hotfix/MainActivity.kt
package com.imooc.hotfix

import android.Manifest
import android.content.pm.PackageManager
import android.os.Build
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.core.app.ActivityCompat

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        if (Build.VERSION.SDK_INT > Build.VERSION_CODES.LOLLIPOP) {

            if (ActivityCompat.checkSelfPermission(
                    this,
                    Manifest.permission.READ_EXTERNAL_STORAGE
                ) != PackageManager.PERMISSION_GRANTED) {

                ActivityCompat.requestPermissions(
                    this,
                    arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE),
                    1
                )

            }

        }

        // 设置点击事件
        findViewById<View>(R.id.btn).setOnClickListener {

            // 弹出一个Toast，显示 ToBeFixed 这个类返回的字符串
            Toast.makeText(it.context, ToBeFixed.get(), Toast.LENGTH_LONG).show()
        }
    }
}